dockstore-client/src/main/java/io/dockstore/client/cli/nested/WorkflowClient.java

```diff
@@ -663,9 +664,29 @@ protected void handlePublishUnpublish(String entryPath, String newName, boolean
                     publish(true, entryPath);
                 }
             } else {
-                //for workflows method currently doesn't work with --entryname flag
-                errorMessage("Parameter '--entryname' not valid for workflows. See `workflow publish --help` for more information.",
-                    CLIENT_ERROR);
+                try {
+                    final Workflow workflow = workflowsApi.getWorkflowByPath(entryPath, null, false);
+
+                    // path should be represented as repository organization and name (ex. dockstore/dockstore-ui2)
+                    final Workflow newWorkflow = workflowsApi.manualRegister(
+                            getGitRegistry(workflow.getGitUrl()),
+                            workflow.getOrganization() + "/" + workflow.getRepository(),
+                            workflow.getWorkflowPath(),
+                            newName,
+                            workflow.getDescriptorType().toString(),
+                            workflow.getDefaultTestParameterFilePath()
+                    );
+
+                    if (newWorkflow != null) {
+                        out("Successfully registered " + entryPath + "/" + newName);
+                        workflowsApi.refresh(newWorkflow.getId());
```

Comment:
Is this line necessary? I think it is, but just double-checking.

Reply:
Ideally, manualRegister will not return a null value, and throw the expected API exception. In the theoretical case it doesn't, this is extra verbose.